client/src/components/SiteSearch/SiteSearchSidebar/SiteSearchSidebar.scss

```diff
@@ -46,3 +46,7 @@ $core-portal-color-primary: var(--global-color-accent--normal);
   flex-grow: 1;
   justify-content: flex-end;
 }
+
+.hidden {
+  visibility: hidden;
+}
```

Comment:
Does the relevant UI use and rely on the feature of `visibility: hidden` to allow element to still take up space? (Seems not when I hastily tested Data Files. I can not test Site Search locally.) If not, consider `hidden` attribute i.e. `<div className="search-count-pill" hidden={searching} ...>`.[^1]

[^1]: https://css-tricks.com/inclusively-hidden/ at "Hidden for everyone?"[^2]
[^2]: We don't have [Monica D's issue](https://meowni.ca/hidden.is.a.lie.html), but Bootstrap already has `[hidden]{display:none!important}` so we'd be safe.